### PR TITLE
Fixed Migrate site switcher to use Stimulus approach AutoFieldController #10035

### PR DESCRIPTION
**Pull request by Aadijindal07**
Solution for issue #10035

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [x] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [x] **Please list the exact browser and operating system versions you tested**:
    -   [x] **Please list which assistive technologies [^3] you tested**:
-   [x] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by lb-** [comment]
Solution for #10035

@Aadijindal07 please ensure you include the issue number being fixed in the PR description. 

We have some recommended reading for new contributors here https://wagtail.org/blog/ten-tasty-ingredients-for-a-delicious-pull-request/

It's good to take the time to read this as it helps with ensuring your PR is good to review.

**Review by lb- — CHANGES_REQUESTED** [review]
@Aadijindal07  can you confirm you tested this?

It does not work at all sorry, there might be some confusion on the issue that we should clarify but I tested this manually and it's not triggering any URL change on select.

Screenshot 2023-02-07 at 10 17 22 pm

File: client/src/controllers/AutoFieldController.ts [hunk 1]
```diff
@@ -24,6 +24,12 @@ export class AutoFieldController extends Controller<
       );
     }
 
+    // Update form action based on selected option
+    if (this.element.type === 'select-one') {
+      const selectedOption = this.element.options[this.element.selectedIndex];
+      form.action = selectedOption.dataset.action;
+    }
+
```
Comment (lb-) [inline client/src/controllers/AutoFieldController.ts:32]:
```suggestion
    if (this.syncActionValue) {
      form.action = this.element.value;
    }
```

Note: this assumes you are going to use a Stimulus value approach, which will need to be set up above as follows:

```
// ...
  HTMLInputElement | HTMLSelectElement
> {
  static values = {
    /** When true, update the form action URL to match the value provided */
    syncAction: { default: false, type: Boolean },
  };

  declare syncActionValue: boolean;
```

File: wagtail/contrib/settings/forms.py [hunk 2]
```diff
@@ -5,17 +5,8 @@
...
-    def media(self):
-        return forms.Media(
-            js=[
-                versioned_static("wagtailsettings/js/site-switcher.js"),
-            ]
-        )
+    site = forms.ChoiceField(choices=[], widget=forms.Select(attrs={'data-auto-submit': ''}))
 
     def __init__(self, current_site, model, **kwargs):
         initial_data = {"site": self.get_change_url(current_site, model)}
```
Comment (lb-) [inline wagtail/contrib/settings/forms.py:12]: Note: There will still need to be a solution for ensuring that the 'current' site is disabled or somehow set to not trigger a page reload.

Maybe that logic could be in the controller, but ideally we can find a more semantic way to do this in the HTML.

**Review by lb- — COMMENTED** [review]
Some tips to help you get this moving.

**Comment by Aadijindal07** [comment]
@lb- thanks a lot brother

**Comment by lb-** [comment]
@Aadijindal07 please note that the `AutoFieldController` has now been renamed to `SubmitController`. You will need to rebase these changes or potentially redo some of them.

While normally we recommend not doing a new PR for changes on the same 'fix', you may be better off actually creating a new branch off of `main` in this case if you get stuck.

#10035

> Now that #10098 is in - I have renamed and refined this issue description to reflect `w-submit` / `SubmitController`.

**Comment by lb-** [comment]
@Aadijindal07  just checking in if you are still ok to finish this out, it's ok if not.

Note that we are doing a webinar about contributing to the Stimulus adoption next week that may help.

> This is happening next Tuesday, 28th of Feb. More details and links to join: https://wagtail.org/blog/contributors-webinar-plans-for-stimulus-in-wagtail/

**Comment by Aadijindal07** [comment]
@lb- Sorry bro I was inactive actually exam are going on and gsoc also. I will definitely do it in a week's time. Btw please give some suggestion for gsoc any useful tip or something as am targeting wagtail as it fits my interest. 
Regards
Aadi 
✌🏻

**Comment by lb-** [comment]
Thanks for the update. Don't want to pressure - just wanted to check in. 

I think for GSoC - best to.char about it in the channel on Slack. This kind of code though will give you a good sense of what it's like writing some JS in Wagtail.

**Comment by Aadijindal07** [comment]
@lb- thanks a lot brother 😉

**Review by lb- — COMMENTED** [review]
@Aadijindal07  as we are close to the 5.0 release candidate, I have added the suggested changes and unit tests.

After testing I found that the GET form submit will not work so have changed to a redirect, but we did not need to worry about the disabled state of the selected site. As if you simply re-select the current value, no `change` event is fired.

I will push up this change soon and confirm the CI passes and merge.

Thanks so much for putting this original PR together.

**Review by lb- — APPROVED** [review]

**Comment by lb-** [comment]
Rebased after #10212 & moved the behaviour to `w-action`, same code/tests but the action controller makes more sense here as we are no longer actually submitting the existing form.

File: wagtail/contrib/settings/forms.py (hunk 2 above)
Comment (lb-) [reply wagtail/contrib/settings/forms.py:20]: As per other comment, this ended up not being required.

**Pull request merged**